### PR TITLE
fix: only get stack info on delete if deleting images

### PR DESCRIPTION
**Pull request by jakeyheath**
<details open>
  <summary><a href="https://czi-tech.atlassian.net/browse/CCIE-2231" title="CCIE-2231" target="_blank">CCIE-2231</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
  <td>Delete is bailing too quickly when state isn't there</td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Task
        Task
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>In Progress</td>
    </tr>
  </table>
</details>

---

## Summary

This fixes a bug where the stacks couldn't be deleted if there was no statefile yet for the stack. This was due to a feature flag added. This fix moves the logic for the feature flag so that delete's no longer accidently bail when deleting stacks without statefiles.

### CODE REVIEW

File: cli/go.mod [hunk 1]
```diff
@@ -1,4 +1,6 @@
-go 1.20
+go 1.21
```
Comment (hspitzley-czi) [inline cli/go.mod:1]: Is this upgrade intended as part of this?

Comment (hspitzley-czi) [reply cli/go.mod:1]: I would advocate for taking this on later and doing all happy packages at the same time

Comment (jakeyheath) [reply cli/go.mod:1]: not sure how that happened, I will revert

Comment (hspitzley-czi) [reply cli/go.mod]: `gvm` (if you use that) has a tendency to lose track of which go version it's supposed to be on :/

**Review by hspitzley-czi — APPROVED** [review]

**Pull request merged**